### PR TITLE
event: set_task_comm

### PR DESCRIPTION
**Pull request by roikol**
create new event indicating process name change

## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

using `__set_task_comm` kprobe to indicate event of process name change

Fixes: #1810

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

Tests being included in this PR:

- [ ] Test File A
- [ ] Test File B

Reproduce the test by running:

- command 01
- command 02

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [x] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented all functions/methods created explaining what they do.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [x] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

**Review by yanivagman — CHANGES_REQUESTED** [review]
You should probably use task_rename tracepoint which is more stable than this kprobe:
https://elixir.bootlin.com/linux/v5.18.2/source/fs/exec.c#L1231

**Comment by roikol** [comment]
> You should probably use task_rename tracepoint which is more stable than this kprobe: https://elixir.bootlin.com/linux/v5.18.2/source/fs/exec.c#L1231

thanks @yanivagman !
i've updated the PR so now it works with the tracepoint.

File: pkg/ebpf/c/tracee.bpf.c [hunk 1]
```diff
@@ -5391,6 +5392,32 @@ int BPF_KPROBE(trace_security_file_permission)
...
+
+    struct task_struct *tsk = (struct task_struct *)ctx->args[0];
+    char old_name[TASK_COMM_LEN];
+    bpf_probe_read_str(&old_name, TASK_COMM_LEN, tsk->comm);
+    const char *new_name = (const char *)ctx->args[1];
+
+    // this tracepoint is invoked for task renaming - this also happens when a process first starts.
+    // so check if this is new execution or not.
+    // this arg was inspired by __set_task_comm() - https://elixir.bootlin.com/linux/v5.18.2/source/fs/exec.c#L1228.
+    bool exec = (get_task_host_pid(tsk) == get_task_host_tgid(tsk));
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:5412]: Is that always true?
What if the main thread (which has tgid) renamed itself?

Comment (roikol) [reply pkg/ebpf/c/tracee.bpf.c:5412]: your'e right, good point.

i think the way to tackle this is to create a map indicating an exec. this map would be updated in `sched_process_fork`, and deleted from in `sched_process_exec` and `task_rename`.

another is option is to go back to using kprobe `__set_task_comm` (https://elixir.bootlin.com/linux/latest/source/fs/exec.c#L1228) which get this information as an argument.

WDYT?

Comment (yanivagman) [reply pkg/ebpf/c/tracee.bpf.c:5412]: can't we just check if the current syscall is execve/execveat?

Comment (roikol) [reply pkg/ebpf/c/tracee.bpf.c:5412]: yes we can

File: pkg/ebpf/c/tracee.bpf.c [hunk 2]
```diff
@@ -5394,6 +5395,32 @@ int BPF_KPROBE(trace_security_file_permission)
...
+    char old_name[TASK_COMM_LEN];
+    bpf_probe_read_str(&old_name, TASK_COMM_LEN, tsk->comm);
+    const char *new_name = (const char *) ctx->args[1];
+
+    save_str_to_buf(&data, (void *) old_name, 0);
+    save_str_to_buf(&data, (void *) new_name, 1);
+    if (data.options & OPT_SHOW_SYSCALL) {
+        syscall_data_t *sys = bpf_map_lookup_elem(&syscall_data_map, &data.context.host_tid);
+        if (sys) {
+            save_to_submit_buf(&data, (void *) &sys->id, sizeof(int), 2);
```
Comment (yanivagman) [inline pkg/ebpf/c/tracee.bpf.c:5417]: As you now send it as an argument, let's add this to the argument parser to get the syscall name

Comment (roikol) [reply pkg/ebpf/c/tracee.bpf.c:5417]: i've added it to `argprinters.go`. should i do it elsewhere?

Comment (yanivagman) [reply pkg/ebpf/c/tracee.bpf.c:5417]: sorry, I missed that

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**